devtools_m1/cypress/integration/common/admin.js

```diff
@@ -119,14 +122,43 @@ Given('I have configured a configurable widget for website {string}', function(s
           "name": "Widget 1 XL",
           "description": "This is really an XL widget. There are many like it, but this one is mine.",
           "shortDescription": "This is really an XL widget.",
-          "websiteIds": [websiteId],
+          "websiteIds": [websiteId]
         },
         "L": {
           "sku": "widg-1-l",
           "name": "Widget 1 L",
           "description": "This is really an L widget. There are many like it, but this one is mine.",
           "shortDescription": "This is really an L widget.",
-          "websiteIds": [websiteId],
+          "websiteIds": [websiteId]
+        }
+      }
+    }
+  })
+})
+
+Given('I have configured a configurable widget with an invisible child', function() {
+  cy.createProduct({
+    "sku": "widg-1",
+    "name": "Widget 1",
+    "description": "This is really a widget. There are many like it, but this one is mine.",
+    "shortDescription": "This is really a widget.",
+    "image": "parent_image.png",
+    "typeId": "configurable",
+    "child_visibility": 4, // not visible in either seach or catalog
```

Comment:
Another approach that wouldn't require any code changes in the product builder would be to add `"visibility":4` to each child description below.

Reply:
I thought that would get overridden here: https://github.com/DripEmail/magento-m1-extension/blob/master/devtools_m1/php_utils/create_product.php#L71. Is that not the case?

Reply:
Heh, good point. I think that the fix for that is to not special-case visibility:

```diff
diff --git a/devtools_m1/php_utils/create_product.php b/devtools_m1/php_utils/create_product.php
index 37738d7..b5a2eb6 100644
--- a/devtools_m1/php_utils/create_product.php
+++ b/devtools_m1/php_utils/create_product.php
@@ -59,6 +59,7 @@ class Mage_Shell_Drip_CreateProduct extends Mage_Shell_Abstract
                 "is_in_stock" => 1,
                 "qty" => 999
             ),
+            "visibility" => Mage_Catalog_Model_Product_Visibility::VISIBILITY_BOTH, //catalog and search visibility
         );
         $fullData = array_replace_recursive($defaults, $data);
 
@@ -68,8 +69,6 @@ class Mage_Shell_Drip_CreateProduct extends Mage_Shell_Abstract
             $product->$methodName($value);
         }
 
-        $product->setVisibility(Mage_Catalog_Model_Product_Visibility::VISIBILITY_BOTH); //catalog and search visibility
-
         return $product;
     }
 

```

Reply:
Hmm. Before you made that comment, I wondered if I was missing something and tried setting visibility as you suggested. Tests passed. So maybe that doesn't override after all?